web-ui/jsoo/main.ml

```diff
@@ -0,0 +1,80 @@
+module Dom_html = Js_of_ocaml.Dom_html
+module Dom = Js_of_ocaml.Dom
+module Js = Js_of_ocaml.Js
+module Firebug = Js_of_ocaml.Firebug
+
+(* module WebSockets = Js_of_ocaml.WebSockets *)
+module WebSocket = Brr_io.Websocket
+module Ev = Brr.Ev
+
+let regexp_left_paren = Re.Str.regexp_string "("
+let regexp_right_paren = Re.Str.regexp_string ")"
+let encoded_left_paren = "%28"
+let encoded_right_paren = "%29"
+
+let encode_parens s : Jstr.t =
+  let s' = Jstr.to_string s in
+  let s_lp_encoded =
+    Re.Str.global_replace regexp_left_paren encoded_left_paren s'
+  in
+  let s_rp_encoded =
+    Re.Str.global_replace regexp_right_paren encoded_right_paren s_lp_encoded
+  in
+  Jstr.of_string s_rp_encoded
+
+(* let js_str = Js.string *)
+let document = Dom_html.document
+
+let inject_log_lines data =
+  let new_div = Dom_html.createDiv document in
```

Comment:
At least on Safari, adding this extra div around the logs creates ugly spacing between the log lines. I'm unsure why.

<img width="338" alt="Screenshot 2023-03-22 at 11 49 44" src="https://user-images.githubusercontent.com/13054139/226882731-d6909295-2f2c-40a2-b706-16ec78e9ddb9.png">

Reply:
Wondering whether this has to do with the padding that is added to the first and last child ... might try taking that out and solving for neatness differently (like adding a margin on the container).

Reply:
I think I have fixed this now @benmandrew - would you please run up my branch locally and verify that it renders well for you in Safari?